src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -168,7 +168,7 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		}
 		decoration.onRender(element => {
 			decoration.onDispose(() => this._decorations.delete(decoration.marker.id));
-			if (beforeCommandExecution && !this._placeholderDecoration) {
```

Comment:
`this._placeholderDecoration` was still defined (though disposed of), so the else was happening and a decoration was getting added

Reply:
Should we have this here too?

```ts
if (!this._placeholderDecoration.isDisposed) {
  this._placeholderDecoration.dispose();
}
```

Should probably have a helper to do that for us in order to prevent future issues like this?